packages/x-adapter/README.md

```diff
@@ -1,13 +1,218 @@
+interface ApiRequestModel {
+  q: string;
+  limit?: number;
+  offset?: number;
+}
+interface ApiResponseModel {
+  items: T[];
+  limit?: number;
+  offset?: number;
+  total: number;
+}
+
+// Maps both the request and the response to connect your model with API you are working with.
+const searchRequestSchema: Schema<SearchRequest, ApiRequestModel> = {
+  q: 'query',
+  limit: 'rows',
+  offset: 'start'
+};
+const searchResponseSchema: Schema<ApiResponseModel, SearchResponse> = {
+  results: 'items',
+  totalResults: 'items.total' // you can use paths to access to an inner property if needed
+};
+
+// The 'schemaMapperFactory' creates a mapper function for a given schema.
+export const searchRequestMapper = schemaMapperFactory<SearchRequest, ApiRequestModel>(
+  searchRequestSchema
+);
+export const searchResponseMapper = schemaMapperFactory<ApiResponseModel, SearchResponse>(
+  searchResponseSchema
+);
+```
+
+#### Create more complex models with Subschemas
+
+For more complex data mapping, the `x-adapter` library also provides a `subSchema` type that
+receives a `Source`, a `Target` and a `Path` that will be applied to the inner path of the main
+object. Furthermore, and not restricted to using subschemas, you can also use schema with function
+return types, that makes schemas powerful to custom the logic you need in your data mapping.
+
+```tsx
+// Example of using a subSchema to map a response
+const responseMapper = schemaMapperFactory<ApiResponseModel, SearchResponse>({
+  totalResults: 'results.total',
+  results: {
+    $path: 'results.items',
+    $subSchema: {
+      id: 'id',
+      name: 'name',
+      url: 'result_url',
+      images: ({ result }) => result.images.map(({ url }) => url),
+      price: {
+        value: ({ price, sale_price }) => (sale_price ? sale_price : price)
+      }
+    }
+  }
+});
+```
+
+#### Using a mutable schema
+
+The `x-adapter` library also provides a `MutableSchema` type, a `Schema` that can replace an
+original schema, override it or create a new one based on the original one. This can be useful to
+define a shared pattern that you will need to reuse in different endpoint adapters.
+
+```tsx
+import { createMutableSchema, Schema } from '@empathyco/x-adapter';
+import { SearchRequest } from '@empathyco/x-types';
+import { ApiRequestModel } from 'YOUR-API-SOURCE';
+
+export const searchRequestSchema = createMutableSchema<Schema<SearchRequest, ApiRequestModel>>({
+  q: 'query',
+  limit: 'rows',
+  offset: 'start'
+  // ... any existing types from your API source
+});
+```
+
+### Extend an adapter that uses schemas
+
+You can check the
+[x-platform-adapter](https://github.com/empathyco/x/tree/main/packages/x-adapter-platform) library.
+You will find a sample implementation of the `x-adapter` library based on the
+[Search Platform API](https://docs.empathy.co/develop-empathy-platform/api-reference/search-api.html),
+and also some guidance on how to extend it for your needs.
+
+## Test
```

Comment:
I would add a little introduction, explaining that we use `jest` for the tests and where are they located.